native/NetworkListener/MessageManager.cpp

```diff
@@ -0,0 +1,556 @@
+﻿#include "stdafx.h"
+#include "MessageManager.h"
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <locale>
+#include <codecvt>
+#include <cwctype>
+#include <collection.h>
+#include <ppltasks.h>
+
+using namespace std;
+using namespace Concurrency;
+using namespace Platform::Collections;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+using namespace Windows::Storage::Streams;
+using namespace Windows::Web::Http;
+using namespace Platform;
+using namespace NetworkProxyLibrary;
+
+MessageManager::MessageManager(unsigned int processId)
+{
+    _processId = processId;
+    _currentMessageCounter = 1;    
+    _requestSentDictionary = ref new Map<Guid, JsonObject^>();
+    _retryQueue = ref new Vector<Message^>();
+}
+
+MessageManager::~MessageManager()
+{
+}
+
+String^ MessageManager::GetNextSequenceId(IdTypes counterType)
+{
+    // restart counter when max number is achieved
+    if (_idCounters[counterType] == INT_MAX)
+    {
+        _idCounters[counterType] = 1;
+    }
+    return _processId.ToString() + "." + _idCounters[counterType]++;
+}
+
+void InsertString(JsonObject^ json, String^ key, String^ value)
+{
+    json->Insert(key, JsonValue::CreateStringValue(value));
+}
+
+void InsertNumber(JsonObject^ json, String^ key, double  value)
+{
+    json->Insert(key, JsonValue::CreateNumberValue(value));
+}
+
+void MessageManager::SendToProcess(Message^ message) 
+{
+    try
+    {
+        OutputDebugStringW(L"Enter SendToProcess \n");                                     
+        this->ProcessMessage(message);                                          
+        OutputDebugStringW(L"Exit SendToProcess \n");
+    }
+    catch (const std::exception& ex)
+    {
+        auto t = ex;
```

Comment:
I have removed the try/catch, it was there by mistake. I had put it there while I was debugging at the beginning some errors and I forgot to remove it. It should not be there.